Server/Services/LeaderBoardService.cs

```diff
@@ -0,0 +1,114 @@
+        /// It gets winning teams, losing teams and then updates the new rating
+        /// </summary>
+        /// <param name="request"></param>
+        /// <param name="players"></param>
+        /// <returns></returns>
+        public async Task<List<Player>> UpdateNewRating(AfterMatchRequest request, List<Player> players)
+        {
+            var winningTeam = GetWinningTeam(request, players, out var averageWinningRating);
+            var losingTeam = GetLosingTeam(request, players, out var averageLosingRating);
+            var scoreDelta = CalculateScoreDelta(averageWinningRating, averageLosingRating);
+            await UpdateRating(winningTeam, losingTeam, scoreDelta);
+            return players;
+        }
+
+
+        /// <summary>
+        /// Formula is the difference between loosing and winning team of avg rating divided by 40
+        /// </summary>
+        /// <param name="averageWinningRating"></param>
+        /// <param name="averageLosingRating"></param>
+        /// <returns></returns>
+        private int CalculateScoreDelta(double averageWinningRating, double averageLosingRating)
+        {
+            var scoreDeltaDouble =
+                -(averageWinningRating - averageLosingRating) /
+                _divisionPoints; // e.g. - (1900 - 2100) / 40 = 5. Meaning winning team will get 5 more points 
+                                // than the base as their average was weaker
+            var scoreDelta =
+                Convert.ToInt32(Math.Round(scoreDeltaDouble, 0,
+                    MidpointRounding.AwayFromZero)); // we don't do a conversion straight to the double
+                                                     // as it will do a MidpointRounding.ToEven by default
+            scoreDelta = scoreDelta > _maximumDelta ? _maximumDelta : scoreDelta;
+            return scoreDelta;
+        }
+
+        private IEnumerable<Player> GetLosingTeam(AfterMatchRequest request, List<Player> allPlayers, out double averageLosingRating)
+        {
+            var losingTeamIds = request
+                .Players
+                .Where(p => p.Team != request.Winner)
+                .Select(p => ulong.Parse(p.SteamId))
+                .ToList();
+            var losingTeam = allPlayers.Where(p => losingTeamIds.Contains(p.SteamId));
+            averageLosingRating = losingTeam.Average(p => p.Rating12v12);
+            return losingTeam;
+        }
+
+        private IEnumerable<Player> GetWinningTeam(AfterMatchRequest request, List<Player> allPlayers, out double averageWinningRating)
+        {
+            var winningTeamIds = request
+                .Players
+                .Where(p => p.Team == request.Winner)
+                .Select(p => ulong.Parse(p.SteamId))
+                .ToList();
+            var winningTeam = allPlayers.Where(p => winningTeamIds.Contains(p.SteamId));
+            averageWinningRating = winningTeam.Average(p => p.Rating12v12);
+            return winningTeam;
+        }
```

Comment:
These are too close to be duplicated. One option is to make the kind a parameter, another is to process both winners and losers in a single method (splitting it with a partition) and just return both winners and losers as a tuple